core/server/common/src/main/java/alluxio/master/journal/raft/SnapshotReplicationManager.java

```diff
@@ -449,95 +463,67 @@ private void requestSnapshotFromFollowers() {
                   .newBuilder()
                   .setSnapshotInfoRequest(GetSnapshotInfoRequest.getDefaultInstance())
                   .build()))));
+      // query all secondary masters for information about their latest snapshot
       for (Map.Entry<RaftPeerId, CompletableFuture<RaftClientReply>> job : jobs.entrySet()) {
         RaftPeerId peerId = job.getKey();
-        RaftClientReply reply;
-        try {
-          reply = job.getValue().get();
-        } catch (Exception e) {
-          LOG.warn("Exception thrown while requesting snapshot info {}", e.toString());
-          continue;
-        }
-        if (reply.getException() != null) {
-          LOG.warn("Received exception requesting snapshot info {}",
-              reply.getException().getMessage());
-          continue;
-        }
-        JournalQueryResponse response;
         try {
-          response = JournalQueryResponse.parseFrom(
+          RaftClientReply reply = job.getValue().get();
+          if (reply.getException() != null) {
+            throw reply.getException();
+          }
+          JournalQueryResponse response = JournalQueryResponse.parseFrom(
               reply.getMessage().getContent().asReadOnlyByteBuffer());
-        } catch (InvalidProtocolBufferException e) {
-          LOG.warn("Failed to parse response {}", e.toString());
-          continue;
-        }
-        LOG.debug("Received snapshot info from follower {} - {}", peerId, response);
-        if (!response.hasSnapshotInfoResponse()) {
-          LOG.warn("Invalid response for GetSnapshotInfoRequest {}", response);
-          continue;
-        }
-        SnapshotMetadata latest = response.getSnapshotInfoResponse().getLatest();
-        if (latest == null) {
-          LOG.debug("Follower {} does not have a snapshot", peerId);
-          continue;
-        }
-        if (snapshotMetadata == null
-            || (latest.getSnapshotTerm() >= snapshotMetadata.getSnapshotTerm())
-            && latest.getSnapshotIndex() > snapshotMetadata.getSnapshotIndex()) {
-          snapshotMetadata = latest;
-          snapshotOwner = peerId;
+          LOG.debug("Received snapshot info from follower {} - {}", peerId, response);
```

Comment:
I'd put this after the below check. It misguided me while investigating to think the snapshot info was received and ok.